### PR TITLE
Typescript migration (closes #16)

### LINKED ISSUE #16: Use ES6 imports and TypeScript
Let's add TypeScript to our project and use `import`/`export` instead of `require`/`module.exports`. 

I don't insist that you should transform all your JS files to TS files right now. Just try setting `allowJS: true` & `checkJs: false` and compile your JS files without type checking.

### PR DESCRIPTION
**Pull request by Ogurecher**

### CODE REVIEW

**Comment by Ogurecher** [comment]
I created this branch on top of the previous one (before it was merged) and it seems to have dragged the old commits with it...

**Comment by AndreyBelym** [comment]
Yep, looks like we need to rebase it. About ESLint: https://github.com/Ogurecher/Team-Link/issues/18#issuecomment-588059349

File: package.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
 {
   "name": "team-link",
   "version": "1.0.0",
-  "description": "Provides convenient interface to call all online team members via MS Teams",
-  "main": "src/index.js",
+  "description": "Provides a convenient interface to call all online team members via MS Teams",
+  "main": "src/index.ts",
```
Comment (AndreyBelym) [inline package.json]: We shouldn't use TS files as an entrypoint. Your users won't be able to execute it without `ts-node`, it should point to the transpiled files. Ideally, we shouldn't distribute our TS sources at all, only transpiled JS files and generated `.d.ts` definition files.

File: src/Server.ts [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
+    port
+    staticPath
+    server
+
+    constructor ({ app, port, host, staticPath }) {
         this.app = app;
         this.host = host;
         this.port = port;
-        this.staticPath = path.join(__dirname, staticPath);
+        this.staticPath = path.join(path.resolve(), staticPath);
```
Comment (AndreyBelym) [inline src/Server.ts]: You can simplify it with `path.resolve(staticPath)`.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "compilerOptions": {
+        "outDir": "dist",
+        "allowJs": true,
```
Comment (AndreyBelym) [inline tsconfig.json]: Well, since you converted all your JS files to TS, we don't need this flag.

File: src/Server.ts (hunk 2 above)
Comment (Ogurecher) [reply src/Server.ts]: That does not work as intended:
path.join(path.resolve(), staticPath) -> D:\Training\Team-Link\resources
path.resolve(staticPath) -> D:\resources

Comment (Ogurecher) [reply src/Server.ts]: path.resolve(staticPath) now works (staticPath should not start with '/')

File: package.json [hunk 4]
```diff
@@ -1,14 +1,15 @@
 {
   "name": "team-link",
   "version": "1.0.0",
-  "description": "Provides convenient interface to call all online team members via MS Teams",
-  "main": "src/index.js",
+  "description": "Provides a convenient interface to call all online team members via MS Teams",
+  "main": "dist/src/index.js",
```
Comment (AndreyBelym) [inline package.json]: We use `lib` to store transpiled files, including the client scripts.

File: package.json [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
   "scripts": {
-    "start": "node src/index.js",
-    "lint": "eslint src test",
-    "test": "npm run lint && mocha test/**/*-test.js",
+    "start": "npm run build && node dist/src/index.js",
+    "lint": "eslint src/**/*.ts test/**/*-test.ts",
```
Comment (AndreyBelym) [inline package.json]: It's better to use `--ext .ts` for this.

File: resources/client_scripts/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+    "compilerOptions": {
```
Comment (AndreyBelym) [inline resources/client_scripts/tsconfig.json]: We can reuse the root `tsconfig.json` with `extends`.

File: src/App.ts [hunk 7]
```diff
@@ -1,14 +1,17 @@
...
+import Config from './Config';
 
 const configInstance = new Config();
 const config = configInstance.config();
 const info = debug('team-link:info');
 const error = debug('team-link:error');
 
-class App {
+export default class App {
+    app;
```
Comment (AndreyBelym) [inline src/App.ts]: We can add an additional static route to serve client scripts from `lib`.

**Pull request merged**